nab/detectors/numenta/numentaTM_detector.py

```diff
@@ -0,0 +1,68 @@
+# ----------------------------------------------------------------------
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import math
+import simplejson as json
+
+from nupic.algorithms import anomaly_likelihood
+from nupic.frameworks.opf.common_models.cluster_params import (
+  getScalarMetricWithTimeOfDayAnomalyParams)
+from nupic.frameworks.opf.modelfactory import ModelFactory
+
+from nab.detectors.numenta.numenta_detector import NumentaDetector
+
+
+
+class NumentaTMDetector(NumentaDetector):
+  """
+  This detector is derived from NumentaDetector and uses the implementation of temporal memory in https://github.com/numenta/nupic.core/blob/master/src/nupic/algorithms/TemporalMemory.hpp. It differs from its parent detector in temporal memory implementation and its parameters.
+  """
+
+  def __init__(self, *args, **kwargs):
+
+    super(NumentaTMDetector, self).__init__(*args, **kwargs)
+
+  def initialize(self):
```

Comment:
Add a blank line; [we use two blank spaces between methods](https://github.com/numenta/nupic/wiki/Python-Style-Guide#coding-style).